### PR TITLE
add: SECURITY-INSIGHTS.yml

### PR DESCRIPTION
**Pull request by hrittikhere**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
As part of my participation in the Kubernetes Security Slam, It enhance the project's security posture by updating the configuration in the SECURITY-INSIGHTS.yml file.

What problem are you trying to solve?
Enhancing the SECURITY-INSIGHTS.yml configuration will contribute to a more robust security framework for the project, aligning with industry standards and best practices. Related to Kubernetes Security Slam task- 11 and 12.2, mentioned here in the [doc](https://docs.google.com/document/d/1TLahqW6OpWQXbaVJ-SR8z-o0yLSoGKYTHxPU0PyyjXo/edit?pli=1#heading=h.p9rkzz6519c8)

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by eddie-knight** [comment]
In addition to the notes in the description by @hrittikhere, the SECURITY-INSIGHTS.yml file is part of the CNCF CLOMonitor [security hygiene checks](https://clomonitor.io/projects/cncf/gateway-api#gateway-api_security).

File: SECURITY-INSIGHTS.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    ci: true
+    before-release: true 
+security-contacts:
+- type: email
+  value: security@kubernetes.io 
+  primary: true
+vulnerability-reporting:
+  accepts-vulnerability-reports: true
+  email-contact: security@kubernetes.io
+  Security-policy: 'https://kubernetes.io/security/'
```
Comment (eddie-knight) [inline SECURITY-INSIGHTS.yml:49]:
```suggestion
  security-policy: 'https://kubernetes.io/security/'
```

File: SECURITY-INSIGHTS.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  - https://github.com/kubernetes-sigs/gateway-api/blob/main/OWNERS_ALIASES 
+  release-cycle: https://github.com/kubernetes-sigs/gateway-api/blob/main/RELEASE.md
+contribution-policy:
+  accepts-pull-requests: true
+  accepts-automated-pull-requests: true
+  automated-tools-list:
+    - automated-tool: dependabot
+      action: allowed
+      path:
+        - /
```
Comment (eddie-knight) [inline SECURITY-INSIGHTS.yml:21]:
```suggestion
```

`path` is not a required field, so it can safely be omitted when empty

**Review by eddie-knight — COMMENTED** [review]
I recommended one change for @hrittikhere, but beyond that this appears to be a valid file. 

If maintainers have confirmed the accuracy of content, LGTM

**Comment by hrittikhere** [comment]
Thanks @eddie-knight, for the review! Waiting for the maintainers to look into it

File: SECURITY-INSIGHTS.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      action: allowed
+  contributing-policy: https://github.com/kubernetes-sigs/gateway-api/blob/main/CONTRIBUTING.md
+  code-of-conduct: https://github.com/kubernetes-sigs/gateway-api/blob/main/code-of-conduct.md
+documentation:
+  - https://gateway-api.sigs.k8s.io/
+distribution-points:
+  - 'https://pkg.go.dev/sigs.k8s.io/gateway-api'
+security-artifacts:
+  threat-model:
+    threat-model-created: false
```
Comment (robscott) [inline SECURITY-INSIGHTS.yml:28]: Does this count? https://gateway-api.sigs.k8s.io/concepts/security-model/

**Review by robscott — COMMENTED** [review]
Thanks @hrittikhere!

/ok-to-test
/approve

**Pull request closed without merging**